smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/BufferWriter.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.waiters
+
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import java.util.UUID
+
+/**
+ * A class which uses a writer to "record" code into an in-memory buffer,
+ * then "plays it back" onto the writer at a later time.
+ *
+ * Useful when the result of rendering an expression is needed before the expression
+ * is actually to be included in the code.
+ */
+class BufferWriter(
```

Comment:
Can we move this file out of a the waiters folder and to somewhere more suitable for general utility objects?

Reply:
Moved to `software.amazon.smithy.swift.codegen.utils`